builtin/aws/ecs/platform.go

```diff
@@ -1127,305 +1439,756 @@ func (p *Platform) Launch(
 		if aerr, ok := err.(awserr.Error); ok {
 			switch aerr.Code() {
 			case "ResourceConflictException":
-				return nil, err
+				break
 			}
 		}
 
+		s.Update("Failed to register ecs task definition. Will retry in %d seconds (up to %d more times)\nError: %s", awsCreateRetryIntervalSeconds, awsCreateRetries-i, err)
+
 		// otherwise sleep and try again
-		time.Sleep(2 * time.Second)
+		time.Sleep(awsCreateRetryIntervalSeconds * time.Second)
 	}
-
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed registering ecs task definition: %s", err)
 	}
 
 	s.Update("Registered Task definition: %s", family)
 
-	serviceName := fmt.Sprintf("%s-%s", app.App, id)
+	state.Runtime = *runtime
+	state.Arn = *taskOut.TaskDefinition.TaskDefinitionArn
 
-	// We have to clamp at a length of 32 because the Name field to CreateTargetGroup
-	// requires that the name is 32 characters or less.
-	if len(serviceName) > 32 {
-		serviceName = serviceName[:32]
-		L.Debug("using a shortened value for service name due to AWS's length limits", "serviceName", serviceName)
+	s.Done()
+	return nil
+}
+
+func (p *Platform) resourceAlbCreate(
+	ctx context.Context,
+	sg terminal.StepGroup,
+	sess *session.Session,
+	log hclog.Logger,
+	src *component.Source,
+	securityGroups *Resource_ExternalSecurityGroups,
+	subnets *Resource_Subnets, // Required because we need to know which VPC we're in, and subnets discover it.
```

Comment:
I don't love having vpcId live inside the subnet state. I tried having subnets return a "VpcId" marker type, but we're not currently incorporating the resource return values into the argmapper soup. I think this is good enough for now, and if this problem crops up repeatedly we can change the plugin sdk.